app/models/authorization.rb

```diff
@@ -0,0 +1,107 @@
+require 'bcrypt'
+# class Authorization < ActiveRecord::Base
+class Authorization < OmniAuth::Identity::Models::ActiveRecord
+  belongs_to :customer, foreign_key: "customer_id"
+  validates :uid,
+            uniqueness: true,
+            format: {:with => /\A\S+@\S+\z/}, 
+            allow_blank: true,
+            on: :create
+  before_validation do
+    if self.password_digest.blank? && self.password.blank?
+      self.password_digest = String.random_string(6) 
+    end
+  end
+  # validates :password, allow_blank: true, length: { maximum: 20 }
+
+
+  # find or create authorization and customer for non-identity omniauth strategies
+  def self.find_or_create_user auth
+    if user_auth = find_by_provider_and_uid(auth["provider"], auth["uid"])
+      c = user_auth.customer
+    else
+      # create customer
+      c = Customer.new
+      c.email = auth["info"]["email"]
+      customer_name = auth["info"]["name"].split(" ")
+      c.first_name = customer_name[0]
+      customer_name.shift if customer_name[1]
+      c.last_name = customer_name.join("")
+      c = Customer.find_or_create! c
+      # create authorization
+      auth = create :customer => c, :provider => auth["provider"], :uid => auth["uid"]
+    end
+    c
+  end
+
+  def uid
+    if respond_to?("read_attribute")
+      return nil if read_attribute("uid").nil?
+      read_attribute("uid")
+    else
+      raise NotImplementedError 
+    end
+  end
+
+  # create customer and update authorization for omniauth-identity
+  def self.create_user_identity(email, cust_id, password)
+    if auth = find_by(uid: email, provider: nil)
+      # create
+
+      # edge case that an auth was user created with no email given. No way to check this until now, so destroy auth and return error 
+      if email.blank?
+        auth.destroy
+        return nil
+      end
+    
+      # update authorization with new info
+      auth.password = password
+      auth.password_confirmation = password
+      auth.provider = "identity"
+      auth.customer_id = cust_id
+      auth.uid = email
+      auth.save
+    else
+      puts "couldn't find identity"
```

Comment:
That path should never happen. Since you didn't write this part of the code, I wanted to make it as easy as possible to debug potential future problems, but its not necessary at all. I am removing it.